File: javascript/word-count/word-count.js
var Words = function() {};

Words.prototype.count = function(words) {
  var wordCount = Object.create(null);
  var wordList = words.toLowerCase().trim().split(/\s+/);

  for (var i = 0; i < wordList.length; i++) {
    var word = wordList[i];
    if (wordCount[word]) {
      wordCount[word]++;
    } else {
      wordCount[word] = 1;
    }
  }
  return wordCount;
};

module.exports = Words;

File: java/word-count/src/main/java/WordCount.java
import java.util.Map;
import java.util.HashMap;
import java.util.List;

public class WordCount {

    public Map<String, Integer> phrase(String phrase) {
        Map<String, Integer> wordCount = new HashMap();
        String[] words = phrase.replaceAll("[^a-zA-Z0-9 ]", "").split("\\s+");
        for (String word : words) {
            String formattedWord = word.toLowerCase();
            wordCount.put(formattedWord, wordCount.getOrDefault(formattedWord, 0) + 1);
        }
        return wordCount;
    }
}
